### PR TITLE
Add Unit Testing Guidelines to `CONTRIBUTING.md` for Scribe Android

### PR DESCRIPTION
**Pull request by KesharwaniArpita**
### Contributor checklist

-   [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
-   [x] I have tested my code with the `./gradlew lintKotlin detekt test` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Data/blob/main/CONTRIBUTING.md#testing)

---

### Description

This PR enhances the **CONTRIBUTING.md** file by adding comprehensive unit testing guidelines for the Scribe Android project. The added section provides structured guidance on writing, organizing, and naming unit tests, with best practices that will improve test consistency, readability, and effectiveness across the codebase.

### Key Changes:
1. **Importance of Unit Testing**: 
   - A brief overview explaining why unit tests are essential for reliability and maintainability.
   
2. **Project Structure**:
   - Guidelines on where to place test files and new classes for testing, following a consistent folder structure.

3. **Naming Conventions**:
   - Descriptive naming rules for test classes and methods to clearly indicate purpose and expected outcomes.

4. **Scope and Focus**:
   - Emphasis on single-responsibility tests, with setup/teardown instructions for isolated testing.
   - Recommendations on using mocks for dependencies.

5. **Writing Effective Tests**:
   - Using the Arrange-Act-Assert (AAA) pattern to organize test logic clearly.
   - Tips for edge-case coverage and meaningful assertions.

6. **Documentation for Tests**:
   - Guidance on adding comments and failure messages for complex test logic.

### Motivation:
These conventions will help maintain a high standard of testing in the Scribe Android codebase, providing clarity for both current and new contributors and reducing maintenance overhead over time.

CC: @albendz Please review the added conventions and suggest any adjustments if needed.

### Related issue

-   #194

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -159,6 +161,66 @@ ktlint --format
...
+      // Initialize objects
+  }
+
+  @After
+  public void tearDown() {
+      // Cleanup objects
+  }
+  ```
+
+- **Mocking**: Use mocks (e.g., Mockito) to isolate the unit under test, especially with dependencies like databases, network requests, or services.
```
Comment (albendz) [inline CONTRIBUTING.md:206]: This project is setup with MockK so I recommend referring to that instead of Mockito

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -159,6 +161,66 @@ ktlint --format
...
+  - **Assert**: Verify the result.
+
+- **Coverage of Edge Cases**: Write tests for both typical cases and edge cases, like `null` values or invalid data.
+
+#### 5. Test Documentation
+
+- **Comments**: Add comments when test logic is complex or non-intuitive.
+- **Assertions**: Use descriptive assertion methods (`assertTrue`, `assertEquals`, etc.) for clarity and include failure messages for custom assertions if necessary.
+
+Following these conventions ensures consistent, readable, and reliable unit tests, enhancing the quality of the Scribe Android codebase.
```
Comment (albendz) [inline CONTRIBUTING.md:222]: It's good to highlight the importance of testing but this is very similar to the opening content of this section and could be removed. The shorter the docs are, the more likely people are to finish reading through them.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -159,6 +161,66 @@ ktlint --format
 
 <a id="issues-projects"></a>
 
+### Unit Testing Conventions for Scribe Android
+
+To maintain consistency and clarity in the Scribe Android codebase, we recommend you to follow these conventions when writing unit tests. These guidelines cover the organization, naming conventions, scope, and structure for unit tests.
+
+#### 1. Project Structure for Unit Tests
+
+- **Location**: Place all unit tests in the `src/test/java` directory, mirroring the structure of the `src/main/java` directory. For new classes or features, ensure their corresponding test classes follow the same package structure.
```
Comment (albendz) [inline CONTRIBUTING.md:170]: This must be an Android project structure thing - I'm newer to Android - why is kotlin code put under java directories?

The unit tests I added as an example are under src/test/kotlin so a follow up will need to be moving the example to the correct structure if src/test/java is the desired structure.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -159,6 +161,66 @@ ktlint --format
...
+- **Class Files**: Each class in `src/main/java` should have a dedicated test file in `src/test/java`, named by appending `Test` to the class name (e.g., `UserManager` → `UserManagerTest`).
+- **New Classes for Testing**: When a new utility or helper class is needed specifically for testing, place it under `src/test/java/utils` or `src/test/java/helpers`.
+
+#### 2. Naming Conventions for Tests
```
Comment (albendz) [inline CONTRIBUTING.md:174]: Thank you for adding examples in this section

**Review by albendz — COMMENTED** [review]
This covers what I was thinking of when I created the issue - thank you for creating and adding this content!

File: CONTRIBUTING.md (hunk 3 above)
Comment (andrewtavis) [reply CONTRIBUTING.md]: I'm not actually sure on this either, @albendz :) Was the structure of the project that was forked. Maybe we could look into this in #223 where some renaming will happen 🤔

**Review by andrewtavis — APPROVED** [review]
Thanks for the contribution, @KesharwaniArpita :) This will definitely come in handy as we expand out the testing of the project.
